pkg/controller/controller_instance.go

```diff
@@ -1968,7 +1994,7 @@ func (c *controller) checkForRemovedClassAndPlan(instance *v1beta1.ServiceInstan
 	// Regardless of what's been deleted, you can always update
 	// parameters (ie, not change plans)
 	if !isProvisioning && instance.Status.ExternalProperties != nil &&
-		servicePlan.Spec.ExternalName == instance.Status.ExternalProperties.ClusterServicePlanExternalName {
+		servicePlan.Spec.ExternalID == instance.Status.ExternalProperties.ClusterServicePlanExternalID {
```

Comment:
It's not clear to me why this needs to change.

Reply:
It doesn't strictly *need* to change. The External ID of a plan does not change, whereas the External Name of a plan can. So if we rely on comparing the External Names, then we have the potential to send the Plan ID in the update request even when the plan is not changing. If we check the External ID instead, then we will inoculate the check from spurious plan name changes.

Example:
1) Instance is created for plan 1 with name A.
2) Plan 1 has a name change from A to B.
3) After a broker relist, service catalog gets the new name of the plan and changes the external name in the ClusterServicePlan.
4) Instance is updated with new parameters.
5) Controller see that the external plan name stored in ExternalProperties differs from the external plan name from ClusterServicePlan. The Plan ID, which is still 1, is sent with the update request.

Reply:
We use the external plan name currently because that is all we have in ExternalProperties. If the external plan ID have been in ExternalProperties from the start, I would have written the code at that time to use the external plan ID.

Reply:
Good point